secant/Mocked Dependencies/OnboardingMocks.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  OnboardingMocks.swift
+//  secantTests
+//
+//  Created by Francisco Gindre on 9/20/21.
+//
+
+import Foundation
+class OnboardingStepProviderBuilder {
+    var steps: [OnboardingStep] = []
+    var startingAt: Int = 0
+
+    func add(_ step: OnboardingStep) -> Self {
+        steps.append(step)
+        return self
+    }
+
+    func build() -> OnboardingStepProviding {
+        SequencedOnboardingStepProvider(
+            steps: steps
+        )
+    }
+}
+
+extension OnboardingStep {
+    static let stepOne = OnboardingStep(
+        title: "First Onboarding Step",
+        imageName: "figure.wave",
+        blurb: "This is the first step of the Secant Wallet user onboarding",
+        stepNumber: 0
+    )
+
+    static let stepTwo = OnboardingStep(
+        title: "Second Onboarding Step",
+        imageName: "figure.wave",
+        blurb: "This is the Second step of the Secant Wallet user onboarding",
+        stepNumber: 1
+    )
+}
+
+class SequencedOnboardingStepProvider: OnboardingStepProviding {
```

Comment:
Can this be a struct?

Reply:
what's your take on mutating structs?

Reply:
My take is that mutations on a value have easier semantics to reason about than mutations on a reference object.